src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpTraceHelper.cs

```diff
@@ -33,7 +33,27 @@ public static bool IsTraceEnabled()
             return s_TraceEnabled;
         }
 
-        public static void TraceCallbackStatus(string message, IntPtr handle, bool invokeCallback, uint status)
+        public static void Trace(string message)
+        {
+            if (!IsTraceEnabled())
+            {
+                return;
+            }
+            
+            Debug.WriteLine(message);
+        }
+
+        public static void Trace(string format, object arg0, object arg1, object arg2)
```

Comment:
I alluded to this in my previous response before seeing the revamped change, so just commenting on it here again in case the previous comment didn't make sense.  Now the caller that's passing in an IntPtr, a Boolean, and a Boolean is going to be allocating three objects when calling Trace, one for each of the boxed value type arguments, even if IsTraceEnabled is false.  You might just want to guard the call site with a check for IsTraceEnabled, rather than mucking with the Trace signature, e.g.
```C#
if (WinHttpTraceHelper.IsTraceEnabled())
{
    WinHttpTraceHelper.Trace(
        "WinHttpRequestState.Dispose, GCHandle=0x{0:X}, disposed={1}, disposing={2}",
        ToIntPtr(),
        _disposed,
        disposing);
}
```